### PR TITLE
Auto-determine OS and change .env

### PR DESCRIPTION
**Pull request by PavelBulat**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -72,7 +76,17 @@ drush:
 	$(call message,Executing \"drush -r /var/www/html/web $(COMMAND_ARGS) --yes\")
 	$(call docker-www-data, php drush -r /var/www/html/web $(COMMAND_ARGS) --yes)
 
-prepare: | up
+configuring_env:
```
Comment (spleshka) [inline Makefile]: Rename it to something like `prepare\:images`

File: Makefile [hunk 2]
```diff
@@ -72,7 +76,17 @@ drush:
...
+    # If OS Linux - change PHP_TAG in environment.
+	@if [ $(OS_NAME) = "Linux" ]; then \
```
Comment (spleshka) [inline Makefile]: Please use makefile's if/else clause.

**Review by spleshka — DISMISSED** [review]

File: Makefile [hunk 3]
```diff
@@ -72,7 +76,17 @@ drush:
...
+		echo "${cyan}${bold}$(PROJECT_NAME): Defined OS LINUX. Change PHP_TAG for docker image in environment${reset}" \
```
Comment (duozersk) [inline Makefile]: We have a defined function for messages like this, see `$(call message,...)` examples in this file.

File: Makefile [hunk 4]
```diff
@@ -72,7 +76,17 @@ drush:
...
+		# Uncomment all the strings containing 'PHP_TAG'. \
+		sed -i '/PHP_TAG/s/^# //g' \.env; \
+        # Comment all the strings containing 'PHP_TAG' and '-dev-macos-. \
+        sed -i -E '/PHP_TAG.+-dev-macos-/s/^/# /g' \.env; \
```
Comment (duozersk) [inline Makefile]: Smth is wrong with indentations here (indents are not the same for the lines in one block) - check where you use tabs and where the blank symbols. Tabs are essential for Makefile syntax (first tab in a line tells make that it is an instruction to execute / shell command), but once you started the shell command and use the multiline syntax with '\' at the end of lines - then all the indents of the next line are going into this shell command.

File: Makefile [hunk 5]
```diff
@@ -72,7 +76,17 @@ drush:
...
+	fi
+
+prepare: | configuring_env | up
```
Comment (duozersk) [inline Makefile]: There should not be two `|` symbols in this line, only the first one makes sense.

**Review by duozersk — DISMISSED** [review]

File: Makefile [hunk 6]
```diff
@@ -72,7 +76,14 @@ drush:
 	$(call message,Executing \"drush -r /var/www/html/web $(COMMAND_ARGS) --yes\")
 	$(call docker-www-data, php drush -r /var/www/html/web $(COMMAND_ARGS) --yes)
 
-prepare: | up
+prepare\:images:
+    # If OS Linux - change PHP_TAG in environment.
+    ifeq ($(OS_NAME), Linux)
+        $(shell sed -i '/PHP_TAG/s/^# //g' \.env)
```
Comment (spleshka) [inline Makefile]: Need comments what this is doing.

File: Makefile [hunk 7]
```diff
@@ -7,6 +7,10 @@ tests\:prepare tests\:run tests\:cli tests\:autocomplete
 # Create local environment files.
 $(shell cp -n \.\/\.docker\/docker-compose\.override\.default\.yml \.\/\.docker\/docker-compose\.override\.yml)
 $(shell cp -n \.env\.default \.env)
+
+# Define OS system.
```
Comment (spleshka) [inline Makefile]: You don't define it, you save the current OS system value into a varaible.

File: Makefile [hunk 8]
```diff
@@ -72,7 +76,16 @@ drush:
 	$(call message,Executing \"drush -r /var/www/html/web $(COMMAND_ARGS) --yes\")
 	$(call docker-www-data, php drush -r /var/www/html/web $(COMMAND_ARGS) --yes)
 
-prepare: | up
+prepare\:images:
```
Comment (duozersk) [inline Makefile:79]: This new command name should be added at the top in the `.PHONY` commands list.

**Review by duozersk — CHANGES_REQUESTED** [review]
Looks good to me! Please just add a new command name into the `.PHONY` list at the top.

File: Makefile (hunk 8 above)
Comment (PavelBulat) [reply Makefile:79]: Thanks, it is fixed now.

**Review by duozersk — APPROVED** [review]

**Pull request merged**